src/granite_orm/associations.cr

```diff
@@ -1,24 +1,38 @@
 module Granite::ORM::Associations
   # define getter and setter for parent relationship
-  macro belongs_to(model_name)
-    belongs_to {{model_name}}, {{model_name.id}}_id : Int64
+  macro belongs_to(model)
+    {% if model.is_a? TypeDeclaration %}
+      belongs_to {{model.var}}, {{model.type.id}}, {{model.var}}_id : Int64
+    {% else %}
+      belongs_to {{model.id}}, {{model.id.camelcase}}, {{model.id}}_id : Int64
+    {% end %}
   end
 
-  macro belongs_to(model_name, decl)
+  # ditto
+  macro belongs_to(model, decl)
```

Comment:
It looks like `decl` is a pattern that is already used in this file, but what does it stand for? It seems like it means `foreign_key`?